public/js/dashboard.js

```diff
@@ -111,6 +108,8 @@ requirejs([
                 widgetManager.trackPending(kernelFuture, $widgetSubArea.get(0), model);
             });
         });
+    }, function(err) {
+        console.error('Failed to start kernel:', err);
```

Comment:
Is there anything else we want to do here, rather than just log?

Since this is such a serious error, we should probably show an error page/dialog to the user explaining the situation (could be handled as separate issue).